controllers/ControllerDecorator.ts

```diff
@@ -49,6 +49,23 @@ class ControllerRegistration {
     }
 }
 
+class RegistrationHelper {
+    public wildcardCount: number;
+    public segmentCount: number;
+
+    constructor(public routeId: string, public routeUrl: string, public route: RouteRegistration, public middlewares: RequestHandler[]) {
+        this.wildcardCount = this.routeUrl.split('*').length - 1;
+        this.segmentCount = this.routeUrl.split('/').length;
+    }
+}
+
+function wildcardSort(o1: RegistrationHelper, o2: RegistrationHelper): number {
+    if (o1.wildcardCount === o2.wildcardCount) {
+        return 0;
+    }
+    return o1.wildcardCount > o2.wildcardCount ? 1 : -1;
```

Comment:
oh yea, that's the javascript way of compare-functions :-) I did the `C#` variant